### DISCUSSION TITLE
Use trained neural network for different initial conditions

### DISCUSSION THREAD

**Discussion by ZHAIZAHZIAZHL** [lululxvi/deepxde#811] [Q&A]
Dear Dr. Lu Lu

I am working on a project which needs to use a trained neural network to solve ODE problems for different initial conditions. For example, I have a model which could correctly solve the Van der Pol equation with initial conditions y(0)=-2 and dy/dt=-2 at t=0. Now I want to solve the Van der Pol equation with initial conditions y(0)=-1 and dy/dt=-1 at t=0. How could I implement this in Deepxde. Does that mean I have to state new initial conditions and re-compile the model? Since I have re-complied the model, how could I make sure the weight matrix (e.g. dense/kernel:0 from model.print_model()) does not change? I have checked how to re-store the model but I think it can only predict the value based on the "old" initial conditions. 

Best wishes

**Comment by lululxvi** [comment]
You can first restore the weights, and then compile and train the network.

**Reply by ZHAIZAHZIAZHL** [reply]
Thank you for your reply, I have saved the model using model.save() because in the description model.save() will store all variables. Does that mean the weight matrix for new initial conditions will be the same as the previous one? How could I check the weight matrix after the new model has complied before training? 

Many thanks!

**Reply by lululxvi** [reply]
Yes, the same. You can check the prediction.
